optaplanner-core/src/test/java/org/optaplanner/core/api/score/stream/uni/UniConstraintStreamTest.java

```diff
@@ -709,6 +768,11 @@ public void globalNodeOrder() {
 
     @Test
     public void zeroConstraintWeightDisabled() {
+        /*
+         * This may never be possible for Drools. If implemented, please remember that some rules may be shared by
+         * different scoring streams, and therefore the rules can only be disabled when all the relevant scoring streams
+         * have their weights set to zero.
+         */
```

Comment:
This is an important note. That said, I may have an idea how to make the rule disabling work - but we'll have to discuss the impacts. It's possible it would not be reasonable, performance-wise.

Reply:
We need Mario to provide Rule.setEnabled(false).
You can't do this while building the rules (= building the CSSessionFactory), because the constraint weights of @ConstraintConfiguration are part of teh dataset, so they aren't know yet at that moment. However, when the CSSession, the dataset is known, so the weights are known and we can disable the zero weights for that KieSession.

Reply:
@ge0ffrey There is API to remove rules from KieBase. The only question is how to make it not a performance problem to call it when the session is being started.